Sources/NIOSSL/SSLContext.swift

```diff
@@ -451,8 +457,15 @@ extension NIOSSLContext {
             break
         }
     }
+    
+    private static func addCACertificateNameToList(context: OpaquePointer, certificate: NIOSSLCertificate) throws {
+        // Adds the CA name extracted from cert to the list of CAs sent to the client when requesting a client certificate.
+        guard 1 == CNIOBoringSSL_SSL_CTX_add_client_CA(context, certificate.ref) else {
```

Comment:
We have to be a bit careful with lifetimes here: we need to use `withExtendedLifetime(certificate)` around this entire block to make sure we don't accidentally release it.

Reply:
Done.